imports/ui/pages/resources/components/ResourceForm.jsx

```diff
@@ -0,0 +1,288 @@
+import React, { useState, useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import {
+  Box,
+  Button,
+  Center,
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  Switch,
+  Text,
+  VStack,
+  Wrap,
+  IconButton,
+  WrapItem,
+} from '@chakra-ui/react';
+import { SmallCloseIcon } from '@chakra-ui/icons';
+import ReactQuill from 'react-quill';
+import { sortableContainer, sortableElement } from 'react-sortable-hoc';
+import arrayMove from 'array-move';
+import AutoCompleteSelect from 'react-select';
+import makeAnimated from 'react-select/animated';
+
+import { editorFormats, editorModules } from '../../../@/constants/quillConfig';
+import { call, resizeImage, uploadImage } from '../../../@/shared';
+import { message } from '../../../components/message';
+import FormField from '../../../components/FormField';
+import FileDropper from '../../../components/FileDropper';
+import NiceSlider from '../../../components/NiceSlider';
+import Loader from '../../../components/Loader';
+
+const animatedComponents = makeAnimated();
+
+function ResourceForm({ defaultValues, isEditMode, history }) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [resourceLabels, setResourceLabels] = useState([]);
+  const [resourcesForCombo, setResourcesForCombo] = useState([]);
+  const defaultImages = defaultValues?.images ? defaultValues.images : [];
+  const [images, setImages] = useState(defaultImages);
+
+  const { formState, handleSubmit, getValues, register, control } = useForm({
+    defaultValues,
+  });
+  const { isDirty, isSubmitting } = formState;
+  const isCombo = getValues('isCombo');
+
+  const [t] = useTranslation('resources');
+  const [tc] = useTranslation('common');
+
+  useEffect(() => {
+    getResourceLabels();
+    setResourcesForCombo(
+      defaultValues?.resourcesForCombo.map((item) => ({
+        value: item._id,
+        label: item.label,
+      }))
+    );
+  }, []);
+
+  const getResourceLabels = async () => {
+    try {
+      const response = await call('getResourceLabels');
+      setResourceLabels(
+        response.map((item) => ({ value: item._id, label: item.label }))
+      );
+      setIsLoading(false);
+    } catch (error) {
+      message.error(error.reason);
+    }
+  };
+
+  const handleUploadImage = async () => {
+    try {
+      const imagesReadyToSave = await Promise.all(
+        images.map(async (image) => {
+          if (image.type === 'not-uploaded') {
+            const resizedImage = await resizeImage(image.resizableData, 1200);
+            const uploadedImageUrl = await uploadImage(
+              resizedImage,
+              'processDocumentUpload'
```

Comment:
We need to create a new bucket for resource images I think. This is, at says, for process documents...

Reply:
Yes I wasnt sure about how to create a new AWS method, thanx